### PR TITLE
[ext.bridge] Permission decorators and invoke function

### PR DESCRIPTION
**Pull request by Middledot**
## Summary
- A `guild_only` and `has_permissions` decorator that can be used for both app and bridge commands for "frontend" *and* "backend" validation.
- Allow BridgeCommand to be directly invoked (e.g.
```py
@bridge.bridge_command()
async def a_command(self, ctx):
    await other_bridge_cmd.invoke(ctx, ...)
```

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

File: discord/ext/bridge/core.py [hunk 1]
```diff
@@ -376,6 +386,47 @@ def decorator(callback):
...
 
 
+def guild_only():
+    """Intended to work with :class:`.ApplicationCommand` and :class:`BridgeCommand`, adds a :func:`~ext.commands.check`
+    that locks the command to only run in guilds, and also registers the command as guild only client-side (on discord).
+
+    Basically a utility function that wraps both :func:`discord.ext.commands.guild_only` and :func:`discord.commands.guild_only`.
+    """
+    def predicate(callback: Callable):
+        callback.__guild_only__ = True
```
Comment (Dorukyum) [inline discord/ext/bridge/core.py:396]: Shouldn't this check if the passed callback is an instance of `ApplicationCommand` and in that case change the `guild_only` attr instead (without underscores)?
Same for `default_member_permissions`.

File: discord/ext/bridge/core.py [hunk 2]
```diff
@@ -54,6 +57,8 @@
     "BridgeExtGroup",
     "BridgeSlashGroup",
     "map_to",
+    "guild_only",
+    "has_permissions"
```
Comment (Dorukyum) [inline discord/ext/bridge/core.py:61]:
```suggestion
    "has_permissions",
```

**Review by Dorukyum — DISMISSED** [review]

**Comment by BobDotCom** [comment]
@Middledot please review requested changes

**Comment by BobDotCom** [comment]
@Middledot please fix failing tests

**Comment by Middledot** [comment]
> @Middledot please fix failing tests

done

**Review by BobDotCom — DISMISSED** [review]

**Review by BobDotCom — APPROVED** [review]

**Review by ChickenDevs — APPROVED** [review]

**Pull request merged**